.github/workflows/tests_build.yml

```diff
@@ -0,0 +1,147 @@
+---
+name: "Unit tests and docker image build"
+on: push
+env:
+  BASE_IMAGE_NAME: "${{ secrets.DOCKER_ORG }}/geospaas:2.4.0-slim"
+  METANORM_VERSION: '3.1.2'
+jobs:
+  tests:
+    runs-on: 'ubuntu-20.04'
+    env:
+      GEOSPAAS_DB_HOST: 'db'
+      GEOSPAAS_DB_USER: 'test'
+      GEOSPAAS_DB_PASSWORD: ${{ secrets.GEOSPAAS_DB_PASSWORD }}
+      IMAGE_NAME: 'harvesting_tests'
+    steps:
+      - name: 'Checkout repository'
+        uses: actions/checkout@v2
+
+      - name: 'Create test docker network'
+        run: docker network create testing
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASS }}
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-testing-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-testing-
+
+      - name: Build testing image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          target: base
+          build-args: |
+            BASE_IMAGE=${{ env.BASE_IMAGE_NAME }}
+            METANORM_VERSION=${{ env.METANORM_VERSION }}
+          push: false
+          load: true
+          tags: ${{ env.IMAGE_NAME }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new
+
+        # Temp fix
+        # https://github.com/docker/build-push-action/issues/252
+        # https://github.com/moby/buildkit/issues/1896
```

Comment:
necessary? or remained from development?

Reply:
As you can see if you follow the first link, it is a current issue with `build-push-action`.
Without this step the cache grows at each run because old entries are not deleted.